UpdateUserGroups/handler.ts

```diff
@@ -0,0 +1,381 @@
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup
+} from "io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import {
+  AzureUserAttributesMiddleware,
+  IAzureUserAttributes
+} from "io-functions-commons/dist/src/utils/middlewares/azure_user_attributes";
+import {
+  ClientIp,
+  ClientIpMiddleware
+} from "io-functions-commons/dist/src/utils/middlewares/client_ip_middleware";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { EmailAddress } from "../generated/definitions/EmailAddress";
+import { GroupCollection } from "../generated/definitions/GroupCollection";
+import { UserGroupsPayload } from "../generated/definitions/UserGroupsPayload";
+import {
+  getApiClient,
+  IAzureApimConfig,
+  IServicePrincipalCreds
+} from "../utils/apim";
+import { groupContractToApiGroup } from "../utils/conversions";
+import { genericInternalErrorHandler } from "../utils/errorHandler";
+
+type IGetSubscriptionKeysHandlerResponseError =
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+  | IResponseErrorValidation;
+type IGetSubscriptionKeysHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  userAttributes: IAzureUserAttributes,
+  email: EmailAddress,
+  userGroupsPayload: UserGroupsPayload
+) => Promise<
+  | IResponseSuccessJson<GroupCollection>
+  | IGetSubscriptionKeysHandlerResponseError
+>;
+
+function getGroups(
+  apimClient: ApiManagementClient,
+  apimResourceGroup: string,
+  apim: string
+): TaskEither<Error, ReadonlyArray<GroupContract>> {
+  return tryCatch(async () => {
+    // tslint:disable-next-line:readonly-array no-let
+    const groupList: GroupContract[] = [];
+    const groupListResponse = await apimClient.group.listByService(
+      apimResourceGroup,
+      apim
+    );
+    groupList.push(...groupListResponse);
+    // tslint:disable-next-line:no-let
+    let nextLink = groupListResponse.nextLink;
+    while (nextLink) {
+      const nextGroupList = await apimClient.group.listByServiceNext(nextLink);
+      groupList.push(...nextGroupList);
+      nextLink = nextGroupList.nextLink;
+    }
+    return groupList;
+  }, toError);
+}
+
+function getUserGroups(
```

Comment:
isn't this the same code as in `GetUser` ? Can you put these methods into some shared module ?